src/cbc_sdk/endpoint_standard/usb_device_control.py

```diff
@@ -0,0 +1,815 @@
+                          IterableQueryMixin, AsyncQueryMixin)
+from cbc_sdk.errors import ApiError, ServerError
+from cbc_sdk.platform.devices import DeviceSearchQuery
+import logging
+import time
+import json
+
+log = logging.getLogger(__name__)
+
+"""USB Device Control models"""
+
+
+class USBDeviceApproval(MutableBaseModel):
+    """Represents a USB device approval."""
+    urlobject = "/device_control/v3/orgs/{0}/approvals"
+    urlobject_single = "/device_control/v3/orgs/{0}/approvals/{1}"
+    primary_key = "id"
+    swagger_meta_file = "endpoint_standard/models/usbDeviceApproval.yaml"
+
+    def __init__(self, cb, model_unique_id, initial_data=None):
+        """
+        Initialize the USBDeviceApproval object.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            model_unique_id (str): ID of the alert represented.
+            initial_data (dict): Initial data used to populate the alert.
+        """
+        super(USBDeviceApproval, self).__init__(cb, model_unique_id, initial_data)
+        if model_unique_id is not None and initial_data is None:
+            self._refresh()
+
+    @classmethod
+    def _query_implementation(cls, cb, **kwargs):
+        """
+        Returns the appropriate query object for this object type.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            USBDeviceApprovalQuery: The query object for this alert type.
+        """
+        return USBDeviceApprovalQuery(cls, cb)
+
+    def _build_api_request_uri(self, http_method="GET"):
+        """
+        Build the unique URL used to make requests for this object.
+
+        Args:
+            http_method (str): Not used; retained for compatibility.
+
+        Returns:
+            str: The URL used to make requests for this object.
+        """
+        return self.urlobject_single.format(self._cb.credentials.org_key, self._model_unique_id)
+
+    def _refresh(self):
+        """
+        Rereads the object data from the server.
+
+        Returns:
+            bool: True if refresh was successful, False if not.
+        """
+        resp = self._cb.get_object(self._build_api_request_uri())
+        self._info = resp
+        self._last_refresh_time = time.time()
+        return True
+
+    def _update_object(self):
+        """
+        Updates the object data on the server.
+
+        Returns:
+            str: The unique ID of this object.
+        """
+        ret = self._cb.put_object(self._build_api_request_uri(), self._info)
+        return self._refresh_if_needed(ret)
+
+    @classmethod
+    def bulk_create(cls, cb, approvals):
+        """
+        Creates multiple approvals and returns the USBDeviceApproval objects.  Data is supplied as a list of dicts.
+
+        Args:
```

Comment:
Can we include an example of list of approvals given that we don't have a clear picture of the properties of an approval?

```
Example:
   [{...}]
```

Reply:
Adding the example as given in the developer documentation.